### PR TITLE
support  l2_normalize float16

### PR DESCRIPTION
**Pull request by GuoxiaWang**
### PR types

Performance optimization

### PR changes

APIs

### Describe

support  l2_normalize float16

### CODE REVIEW

**Review by Avin0323 — APPROVED** [review]
LGTM for PR-CI-OP-Benchmark.

**Review by TCChenlong — APPROVED** [review]

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5026,64 +5026,42 @@ def l2_normalize(x, axis, epsilon=1e-12, name=None):
...
-	    import paddle.fluid.dygraph as dg
-
-	    with dg.guard(place) as g:
-    		input = dg.to_variable(input_data)
-    		output = fluid.layers.l2_normalize(x=input, axis=-1)
-    		print(output.numpy())
+        
+        X = paddle.randn(shape=[3, 5], dtype='float64')
+        out = paddle.fluid.layers.l2_normalize(X, axis=-1)
+        print(out.numpy())
```
Comment (zhiqiu) [inline python/paddle/fluid/layers/nn.py:5049]: No need `.numpy`, just print tensor

File: python/paddle/fluid/tests/unittests/test_norm_op.py [hunk 2]
```diff
@@ -33,23 +33,27 @@ class TestNormOp(OpTest):
...
         self.attrs = {'epsilon': self.epsilon, 'axis': self.axis}
         self.outputs = {'Out': y, 'Norm': norm}
 
     def test_check_output(self):
-        self.check_output()
+        self.check_output(atol=1e-5)
 
     def test_check_grad(self):
-        self.check_grad(['X'], 'Out')
+        self.check_grad(['X'], 'Out', max_relative_error=0.008)
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/test_norm_op.py:47]: No need to change the max_relative_error of fp64 type.

**Review by zhiqiu — APPROVED** [review]
As discussed, to be refined in the next pr.

**Review by wanghuancoder — APPROVED** [review]

**Pull request merged**